### PR TITLE
Css class list dependency

### PR DESCRIPTION
**Pull request by jouni-kantola**
* Pulled out dependencies
* Renamed modules
* First naive data flow

### CODE REVIEW

File: lib/class-map/index.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+function map(classNames, terseClassNames){
+    if(!Array.isArray(classNames) || !Array.isArray(terseClassNames)) throw new Error("Only arrays are supported mapping types.");
+    
+    return classNames.reduce((prev, curr) => {
+        prev[curr] = terseClassNames.shift();
+        return prev;
+    }, {});
+}
+
+module.exports = {
```
Comment (jouni-kantola) [inline lib/class-map/index.js]: Maybe better to just:
```
module.exports = (classNames, terseClassNames) => {
  ...
}
```
Consume it by:
```const classMap = require("./class-list");
const map = classMap(source, target);```

File: lib/class-map/test/class-map.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            
+            () => {
+                mapper.map("1,2,3", "1,2,3");
+            }.should.throw(expectedError);
+            
+            () => {
+                mapper.map([], "1,2,3");
+            }.should.throw(expectedError);
+            
+            mapper.map([], []).should.be.ok;
```
Comment (jouni-kantola) [inline lib/class-map/test/class-map.js]: Should empty array be OK? Not much use of ruin if it is.

File: lib/class-map/test/class-map.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    describe('map', () => {
+        it('should take list and create map', () => {
+            const classNames = ['a-class', 'another-class'];
+            const terseClassNames = ['a', 'b'];
+            
+            const map = mapper.map(classNames, terseClassNames);
+            const keys = Object.keys(map);
+            keys.forEach((key, index) => {
+                classNames.findIndex(val => val === key).should.equal(index);
+                map[key].should.be.ok;
```
Comment (jouni-kantola) [inline lib/class-map/test/class-map.js]: `terseClassNames` aren't strictly validated.

File: lib/terse-class-list/test/terse-class-list.js [hunk 4]
```diff
@@ -40,5 +40,5 @@
             });
         });
     });
-    
+
```
Comment (jouni-kantola) [inline lib/terse-class-list/test/terse-class-list.js]: What do you like the name `terse-class-list`, @davidxcheng?

Comment (davidxcheng) [inline lib/terse-class-list/test/terse-class-list.js]: I prefer the old name, _terse_ is more like _concise_ to me (ie high information density). The class names are just jibberish and carries no information, they are simply short. Also one might get the impression that the list itself is terse. Makes sense?

Comment (jouni-kantola) [inline lib/terse-class-list/test/terse-class-list.js]: Agree. Let's keep the old name. I'll change it back in the PR.

**Comment by jouni-kantola** [comment]
Updated according to comments. Please have another look, @davidxcheng. :shipit: at will.

**Pull request merged**